daemon/volumes_unix.go

```diff
@@ -270,109 +214,19 @@ func (daemon *Daemon) verifyVolumesInfo(container *Container) error {
 	return nil
 }
 
-// parseVolumesFrom ensure that the supplied volumes-from is valid.
-func parseVolumesFrom(spec string) (string, string, error) {
-	if len(spec) == 0 {
-		return "", "", derr.ErrorCodeVolumeFromBlank.WithArgs(spec)
+// setBindModeIfNull is platform specific processing to ensure the
+// shared mode is set to 'z' if it is null. This is called in the case
+// of processing a named volume and not a typical bind.
+func setBindModeIfNull(bind *volume.MountPoint) *volume.MountPoint {
```

Comment:
okay, I think I didn't express myself correctly, I'm very slow today sorry.

What I'm trying to say is that we should not extract this into a function like this one. I know the check exists already in the unix file. What I'm saying is that we should leave the check as it is now in the unix file and extract only the `"z"` as default mode for unix and probably `""` for windows.

So, instead of replacing:

```go
if bind.Mode == "" {
  bind.Mode = "z"
}
```

with `bind = setDefaultModeIfNull(bind)`, like this PR proposes. We should replace it with:

```go
if bind.Mode == "" {
  bind.Mode = defaultOSBindMode
}
```

where `defaultOSBindMode` is declared in `daemon/volume_unix.go` as:

```go
const defaultOSBindMode = "z"
```

and in `daemon/volume_windows.go` as:

```go
const defaultOSBindMode = "z"
```